Gems/AWSCore/Code/Tools/ResourceMappingTool/resource_mapping_schema.json

```diff
@@ -0,0 +1,56 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema",
+    "type": "object",
+    "title": "O3DE AWS Resource mapping file schema",
+    "required": ["AWSResourceMappings", "AccountId", "Region", "Version"],
+    "properties": {
+        "AWSResourceMappings": {
+            "type": "object",
+            "title": "AWS resource mappings schema",
+            "patternProperties": {
+                "^.+$": {
+                    "type": "object",
+                    "title": "AWS resource entry schema",
+                    "required": ["Type", "Name/ID"],
+                    "properties": {
+                        "Type": {
+                            "$ref": "#/NonEmptyString"
```

Comment:
Nit: Maybe not related to this PR, but we might be able to define pattern for the Type attribute as well. At least all the types seem start with AWS::

Reply:
that is a separate task, feel free to create a task